HTML navigation bars

alurban opened the pull request:
This PR introduces a general-purpose bootstrap navbar constructor, originally designed for [`gwsumm`](https://github.com/gwpy/gwsumm/blob/master/gwsumm/html/bootstrap.py#L89-L157). The following changes are included:

* Introduce a few new functions, including `gwdetchar.io.html.navbar`
* Use this to give navbars to the output of every script, including omega scans
* Move dropdown menu style settings into CSS
* Include unit tests

Test output is available here (all require `LIGO.ORG` credentials):

Tool | Output | Notes
-- | -- | --
`gwdetchar-omega` | [**Link**](https://ldas-jobs.ligo-la.caltech.edu/~aurban/pyomega-test/Network_170817/) | Network scan of GW170817, just to check that the page looks similar
`gwdetchar-scattering` | [**Link**](https://ldas-jobs.ligo-la.caltech.edu/~aurban/scattering/day/20190413/) | L1 scattering on 2019-04-13, full L1 omega scans within
`gwdetchar-overflow` | [**Link**](https://ldas-jobs.ligo-la.caltech.edu/~aurban/overflows/day/20190413/) | L1 DAC overflows on 2019-04-13
`gwdetchar-software-saturations` | [**Link**](https://ldas-jobs.ligo-la.caltech.edu/~aurban/software-saturations/day/20190413/) | L1 software saturations on 2019-04-13
`gwdetchar-lasso-correlation` | [**Link**](https://ldas-jobs.ligo-la.caltech.edu/~aurban/lasso/full-results/day/20190413/) | L1 Lasso run from one of the lock stretches on 2019-04-13

This fixes #288.

cc @duncanmmacleod

duncanmmacleod commented:
@alurban, how do the new scattering, overflow, and software-saturations pages look when embedded inside the summary pages?

alurban commented:
@duncanmmacleod, in [**this example**](https://ldas-jobs.ligo-la.caltech.edu/~aurban/summary/day/20190413/detchar/scattering/), the scattering navbar overtakes the summary page navbar. I think we can handle this in the summary page config files downstream, I'll look into it after lunch.

alurban commented:
@duncanmmacleod, I was right, you can remove the header and footer sections using a fancy regular expression in javascript:

```javascript
var x = data.indexOf('<body');
x = data.indexOf('>', x);
var y = data.lastIndexOf('</body>');
var content = data.slice(x+1, y);
content = content.replace(/(<header.[^]*header>)/g, "");
content = content.replace(/(<footer.[^]*footer>)/g, "");
```

The summary pages will then wrap their own navbar and footer around this, as [**the same example**](https://ldas-jobs.ligo-la.caltech.edu/~aurban/summary/day/20190413/detchar/scattering/) linked to above will show.

tl;dr: we're in the clear as far as the summary pages are concerned.

duncanmmacleod commented on gwdetchar/io/html.py:
```diff
@@ -247,6 +319,238 @@ def new_bootstrap_page(base=os.path.curdir, lang='en', refresh=False,
+    page.li('External', class_='dropdown-header')
+    for name, url in OBSERVATORY_MAP[ifo]['links'].items():
+        if 'Summary' in name:
+            day = str(tconvert(gps).date()).replace('-', '')
```
This might better as
```suggestion
            day = from_gps(gps).strftime(r"%Y%m%d")
```

duncanmmacleod commented on gwdetchar/io/html.py:
```diff
@@ -46,15 +48,85 @@
 from pygments.formatters import HtmlFormatter
 
-from gwpy.time import from_gps
+from gwpy.time import (from_gps, tconvert)
```
```suggestion
from gwpy.time import from_gps
```

alurban commented:
Thanks @duncanmmacleod, I've addressed your review comments above, when build tests pass I'll merge and put out a release as we discussed on this morning's call.
